src/GafferScene/ShaderAssignment.cpp

```diff
@@ -166,6 +168,16 @@ IECore::ConstCompoundObjectPtr ShaderAssignment::computeProcessedAttributes( con
 
 		if( const auto *network = runTimeCast<IECoreScene::ShaderNetwork>( attribute.second.get() ) )
 		{
+			for( const auto &i : network->shaders() )
+			{
+				if( i.second->getName() == ShaderTweakProxy::shaderTweakProxyIdentifier() )
+				{
+					throw IECore::Exception(
+						"ShaderTweakProxy only works with ShaderTweaks - it doesn't make sense to connect one to a ShaderAssignement"
```

Comment:
Seems like either we should be concerned, or we should ditch the TODO about wanting to search for proxies in the input network rather than the network-being-tweaked (in ShaderTweaks).

Reply:
Tried out the optimization discussed this morning. Was a bit surprised I couldn't manage to construct a test where it made more than 10% difference, but still makes sense.

Reply:
Looks like we still have `#include "ShaderTweakProxy.h" in ShaderAssignment.cpp.